### ISSUE TITLE
Rebuild old apps

### ISSUE THREAD

**Issue by btkostner** [elementary/houston#534]
In the short term, if a project hasn't been released in the past month, we should automatically trigger a new build to ensure it still works.

In the long term, we should only do this for apps that need it, like `golang` and `nodejs` apps.

**Comment by cassidyjames** [comment]
What would we do if the app fails, since that version has already been published to the repos?

**Comment by btkostner** [comment]
I would guess we file issues and continue on our way.  Not much we can do past that

**Comment by lewisgoddard** [comment]
Do we de-list the previous build that works (automatically or no) ?

**Comment by btkostner** [comment]
I strongly disagree with removing anything from the repository after it's already been published. The repo should never go backwards as it can lead to some weird package orphan issues.

**Comment by cassidyjames** [comment]
I guess I don't see the point in spinning up rebuilds, then. If a developer is not submitting an update, why rebuild? Is it for security to keep any library changes in sync? In that case, wouldn't this be just as important for Vala apps?

Would we publish the new builds in the repo as a revision?

**Comment by btkostner** [comment]
It's for security, and it would only be for some apps. For instance, not all nodejs packages get security updates from the OS like how a Vala app would.

**Comment by harisvsulaiman** [comment]
@btkostner AFAIK The only nodejs backed app on appcenter is mine, and unfortunately i haven't updated the app in some time ( 3 months ) due to lack of time.

I guess i should integrate https://snyk.io/ , something to stay secure, and patch nodejs often.

@cassidyjames So, Tldr i would definitely support something like what Blake is proposing.

**Comment by harisvsulaiman** [comment]
@btkostner Another option would be to add nodejs lts ppa to elementary os, it comes builtin. (Nodejs/npm is an essential tool for most web devs/ mobile devs right now.)

This could:
 * Continuous security and perf updates. 
 * Decrease my bundle size. ( Would have to change my build configuration)
 * Reduce memory usage of nodejs apps. ( uses 30+ mb  right now, its not a lot 		comparing with electron apps) to 10+mb for small apps.

Would love to hear your thoughts on this.

**Comment by btkostner** [comment]
I don't think including a ppa in the OS is a good idea. It would start with node, and eventually end up including a bunch of third party ppa's that we don't control to accommodate for all the different programming languages.

It would help out with node security, but I am also worried about all the npm package security updates that would not be updated.

**Comment by harisvsulaiman** [comment]
@btkostner fair enough.
Then updating packages thoroughly will be the only way forward.

And integration with something like snyk.

**Comment by harisvsulaiman** [comment]
Wouldn't an app based on python suffer these problems? 

Other packages that are used in the app.(without underlying native libs).

If they have a security hole, should we handle that too.

The only difference I think is nodejs apps depend on nodejs runtime.
We could make it mandatory for apps with a runtime to atleast update the runtime frequently.

**Comment by btkostner** [comment]
I'm not sure about python (I've never programmed in it), but it might be.
